### ISSUE TITLE
`unnecessary_statements` false positive with pattern assignments

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4334]
```dart
var (a, b) = (0, 0);
var result = (1, 2);
(a, b) = (a + result.$1, b + result.$2); // LINTS here (wrongly)
```
